ruma-api-macros/src/util.rs

```diff
@@ -104,13 +161,14 @@ pub(crate) fn build_query_string(request: &Request) -> TokenStream {
             //
             // By asserting that it implements the iterator trait, we can
             // ensure that it won't fail.
-            fn assert_trait_impl<T>()
+            fn assert_trait_impl<T>(_: &T)
             where
                 T: ::std::iter::IntoIterator<Item = (::std::string::String, ::std::string::String)>,
```

Comment:
If we do want to support `query_map`, I guess this needs to be updated from `String` to `&str` (with a lifetime declared on the assert I guess).

Reply:
Or maybe `impl AsRef<str>` actually. But I think it would be better to revert the `query_map` changes for now and leave them for a future PR.

Reply:
The only thing that was changed was `derive_outgoing::strip_lifetimes` got the `Type::Tuple` match for the `Outgoing` trait derive other than that there isn't anything to revert. Shouldn't `strip_lifetimes` handle tuples in general anyway? (this commit https://github.com/ruma/ruma/pull/179/commits/e81e63a8e1ba78c169d050e89ae77e3c7eb3243f) What I'm trying to say is should we leave that so the `Outgoing` derive is more robust, or I could do a new PR just for the `Outgoing` derive.

Reply:
That commit makes sense to keep. It looks like you did another change for query_map a few lines above here though.